jme3-core/src/main/java/com/jme3/scene/control/LightControl.java

```diff
@@ -115,86 +141,133 @@ public void setControlDir(ControlDirection controlDir) {
         this.controlDir = controlDir;
     }
 
-    // fields used when inverting ControlDirection:
+    public Axis getAxisRotation() {
+        return axisRotation;
+    }
+
+    public void setAxisRotation(Axis axisRotation) {
+        this.axisRotation = axisRotation;
+    }
+
+    public boolean isInvertAxisDirection() {
+        return invertAxisDirection;
+    }
+
+    public void setInvertAxisDirection(boolean invertAxisDirection) {
+        this.invertAxisDirection = invertAxisDirection;
+    }
+
+    private void validateSupportedLightType(Light light) {
```

Comment:
I see this no longer allows a null liight and would throw an exception in that case. This could break things for users that are currently setting the light null for some reason. 

There could be situation where someones code intentionally sets the light to null at cleanup, or set null temporarily before being used with a new light. I don't know how probable that is, but its certainly possible and would be an acceptable way to do things that we should support. 

And I suspect that, considering a setLight() method exists, that it is intended that this class should allow the light to be changed at any time. In which case it should also be allowed to be null in cases where there's a delay between 2 lights being used.